pkg/dockerbuilder/kaniko_builder.go

```diff
@@ -0,0 +1,192 @@
+			Get(kanikoJob.Name, meta_v1.GetOptions{IncludeUninitialized: true})
+
+		if err != nil {
+			return errors.Wrap(err, "Failed to poll kaniko job status")
+		}
+
+		if runningJob.Status.Succeeded > 0 {
+			k.logger.Debug("Kaniko job was completed successfully")
+
+			// TODO: Enable cleanup
+			//err = p.consumer.kubeClientSet.BatchV1().Jobs(p.ResolveDefaultNamespace("")).Delete(kanikoJob.Name, nil)
+			//if err != nil {
+			//	p.Logger.Error("Failed to delete Kaniko job after successful completion")
+			//}
+			return nil
+		}
+		if runningJob.Status.Failed > 0 {
+			k.logger.DebugWith("Kaniko job has failed", "status", runningJob.Status)
+			return errors.New("Kaniko job has failed")
+		}
+	}
+	return errors.New("Kaniko job has timed out")
+}
+
+func (k *KanikoBuilder) createDockerBuildBundle(image string, contextDir string, tempDir string) (string, error) {
+	var err error
+
+	// Create temp directory to store compressed docker build bundle
+	buildDockerBundleDir := path.Join(tempDir, "tar")
+	err = os.Mkdir(buildDockerBundleDir, 0744)
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to create tar dir: %s", buildDockerBundleDir)
+	}
+
+	k.logger.DebugWith("Created tar dir", "dir", buildDockerBundleDir)
+
+	tarFilename := fmt.Sprintf("%s.tar.gz", strings.Replace(image, "/", "_", -1))
+	tarFilename = strings.Replace(tarFilename, ":", "_", -1)
+	tarFilePath := path.Join(buildDockerBundleDir, tarFilename)
+
+	k.logger.DebugWith("Compressing build bundle", "tarFilePath", tarFilePath)
+	_, err = exec.Command("tar", "-zcvf", tarFilePath, contextDir).Output()
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to compress build bundle")
+	}
+	k.logger.Debug("Build bundle was successfully compressed")
+
+	// Create symlink to bundle tar file in nginx serving directory
+	err = os.Link(tarFilePath, path.Join("/etc/nginx/static/assets", tarFilename))
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to create symlink to build bundle")
+	}
+
+	return tarFilename, nil
+}
+
+func (k *KanikoBuilder) getJobKanikoSpec(buildOptions *dockerclient.BuildOptions, bundleFilename string) *batch_v1.Job {
+
+	completions := int32(1)
+	buildArgs := []string{
+		fmt.Sprintf("--dockerfile=%s", buildOptions.DockerfilePath),
+		fmt.Sprintf("--context=%s", buildOptions.ContextDir),
+		"--no-push",
+	}
+
+	// TODO: Enable once push to docker registry is sorted out
+	//if !buildOptions.NoCache {
+	//	buildArgs = append(buildArgs, "--cache=true")
+	//}
+
+	// Add build options args
+	for k, v := range buildOptions.BuildArgs {
+		buildArgs = append(buildArgs, fmt.Sprintf("--build-arg=%s=%s", k, v))
+	}
+
+	volumeMount := v1.VolumeMount{
+		Name:      "tmp",
+		MountPath: "/tmp",
+	}
+
+	kanikoJobSpec := &batch_v1.Job{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name: "kanikojob",
+		},
+		Spec: batch_v1.JobSpec{
+			Completions: &completions,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: meta_v1.ObjectMeta{
+					Name: "kanikojob",
```

Comment:
must be unique and contain:
1. the helm release (receive this as an env var under `NUCLIO_DASHBOARD_JOB_NAME_PREFIX`). Obviously you must drag this from the cmd all the way here, you can't access env vars in places other than command parsing
2. the nuclio function name

this goes for both the job and pod spec